app/internal/weekly_tasks.py

```diff
@@ -0,0 +1,138 @@
+        or to edit an existing Weekly Task the user have.
+
+    Args:
+        user (User): The user who wants to add or edit the Weekly Task.
+        session (Session): The session to redirect to the database.
+        mode (str): Determines whether in Add or Edit mode.
+        weekly_task_id (int): In edit mode, represents the
+            weekly task being edited.
+        title (str): Title of the Weekly Task.
+        days (str): Return days of the Weekly Task.
+        content (str): Content of the Weekly Task.
+        is_important (bool): If the task is important.
+        the_time (time): Return time of the Weekly Task.
+
+    Returns:
+        Tuple: Boolean variable which shows if the change has been
+            made to the db, and the model WeeklyTask which the
+            function made so far.
+    """
+    weekly_task = WeeklyTask(
+        title=title,
+        content=content,
+        is_important=is_important,
+        owner_id=user.id
+    )
+
+    if weekly_task_id:
+        weekly_task.id = weekly_task_id
+
+    inputs_ok = check_inputs(days, the_time, title)
+    if not inputs_ok:
+        return False, weekly_task
+
+    weekly_task.days = days
+    weekly_task.the_time = the_time.strftime("%H:%M")
+
+    if mode == "add":
+        user_titles = (
+            user_weekly_task.title
+            for user_weekly_task in user.weekly_tasks
+            )
+        if title in user_titles:
+            return False, weekly_task
+        session.add(weekly_task)
+        session.commit()
+        return True, weekly_task
+
+    else:  # if mode == "edit"
+        w_task_query = session.query(WeeklyTask)
+        old_weekly_task = w_task_query.filter_by(id=weekly_task.id).first()
+
+        user_titles = (
+            user_weekly_task.title for user_weekly_task in user.weekly_tasks
+            if user_weekly_task.title != old_weekly_task.title
+        )
+
+        if title in user_titles:
+            return False, weekly_task
+
+        if old_weekly_task.owner_id != weekly_task.owner_id:
+            return False, weekly_task
+
+        old_weekly_task.title = weekly_task.title
+        old_weekly_task.days = weekly_task.days
+        old_weekly_task.content = weekly_task.content
+        old_weekly_task.is_important = weekly_task.is_important
+        old_weekly_task.the_time = weekly_task.the_time
+        session.commit()
+        return True, weekly_task
+
+
+def make_task(task: Task, user: User, session: Session) -> bool:
+    """Make a task, used by the generate_tasks function"""
+    user_tasks_query = session.query(Task).filter_by(owner_id=user.id)
+    task_by_time = user_tasks_query.filter_by(date_time=task.date_time)
+    task_by_title_and_time = task_by_time.filter_by(title=task.title)
+    task_exist = task_by_title_and_time.first()
+    if not task_exist:
+        session.add(task)
+        session.commit()
+        return True
+    return False
+
+
+def generate_tasks(session: Session, user: User):
```

Comment:
This function does more than one thing. Please split it.

Also, please add a return type.